WordPress/wordpress-playground
# [WordPress build] Only build the latest patch version of WordPress

This PR ensures we always build the latest patch version of each minor WordPress version when running the WordPress build.

Playground supports only the latest patch version of each minor WordPress release (e.g. 6.6, 6.5).
But before this PR, the WordPress build script could include multiple patch versions of WordPress, such as 6.6.2 and 6.6.1.

When this happens the latest version would correctly be the latest minor version (6.6) and point to the latest patch (6.6.2), but the minus one minor version would be the older patch of the latest minor version (6.6.1) instead of the previous minor version (6.5).

This means that a build script like `npx nx bundle-wordpress:major-and-beta playground-wordpress-builds` would correctly build the latest minor version (6.6) with the code from the latest patch (6.6.2). 
After that, instead of building the older minor version (6.5) it would rebuild the latest minor version (6.6) with the code from an older patch (6.6.1).

As a consequence, our git history would be polluted by commits every time this _Refresh WordPress Major&Beta_ workflow runs. 

To address this we filter the list of latest versions to include only the latest patch for each minor version. Older patches are excluded from the list. 

This PR  also prevents the _Refresh WordPress Major&Beta_ workflow from committing changes when the build fails.

## Testing Instructions (or ideally a Blueprint)

- Run ` npx nx bundle-wordpress:major-and-beta playground-wordpress-builds` and confirm there were no changes
